### PR TITLE
[ON-HOLD UNTIL 2020-11-16] Sanlouise 15235 connected apps directory link

### PR DESCRIPTION
**Pull request by sanlouise**
This PR will not be merged until at least November 16th  2020.

## Description
When the user does not have connected apps, we want to show a button that leads the user to an app directory. We also hide the `What other third-party apps can I connect to my profile?` section when the user has no connected apps.

Wen the user does have connected apps, we do still show `What other third-party apps can I connect to my profile?` and do not show the new blue button.

## Testing done
Looks good locally, updated unit tests.

### No connected apps
image

## Acceptance criteria
- [x] Add button to app directory when user has no connected apps
- [x] Update copy under 'What other third party apps can I connect to my profile' dropdown when the user does have connected apps

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by sanlouise** [comment]
@tressaellen Does this look good to you on mobile or should I decrease the button size?
image

**Comment by tressaellen** [comment]
I think that buttons size looks good @sanlouise

**Comment by erikphansen** [comment]
> When the user does have connected apps, we do still show What other third-party apps can I connect to my profile? and do not show the new blue button.

Seems like a weird requirement 🤔

File: src/applications/personalization/profile/components/connected-apps/ConnectedApps.jsx [hunk 1]
```diff
@@ -107,24 +113,12 @@ export class ConnectedApps extends Component {
...
-                      rel="noopener noreferrer"
-                    >
-                      {app.name}
-                    </a>
-                  </li>
-                );
-              })}
-            </ul>
-          </div>
+          <Link
```
Comment (erikphansen) [inline src/applications/personalization/profile/components/connected-apps/ConnectedApps.jsx:116]: Just curious why you are using a `Link` here but previously used `a` elements? Also, does it make a difference if we use the `Link` from old `react-router` like you are doing or if we should use the one from `react-router-dom`?

**Review by erikphansen — APPROVED** [review]

**Comment by sanlouise** [comment]
> > When the user does have connected apps, we do still show What other third-party apps can I connect to my profile? and do not show the new blue button.
> 
> Seems like a weird requirement 🤔

I think the reasoning here is that when the user does not have connected apps, we want to show the user a very clear CTA with the blue button that goes to the app directory - they likely don't know how to connected an app at that point. On the other hand, when the user does have apps connected, this call to action does not need to be prominent.

**Comment by sanlouise** [comment]
Closed in favor of https://github.com/department-of-veterans-affairs/vets-website/pull/15011

**Pull request closed without merging**